python_ta/cfg/graph.py

```diff
@@ -286,29 +286,31 @@ def update_edge_z3_constraints(self) -> None:
                 # traverse through edge
                 if edge.condition is not None:
                     condition_z3_constraint = z3_environment.parse_constraint(edge.condition)
-                    if condition_z3_constraint is not None:
-                        if edge.negate is not None:
-                            z3_environment.add_constraint(
-                                Not(condition_z3_constraint)
-                                if edge.negate
-                                else condition_z3_constraint
-                            )
-
-                edge.z3_constraints[path_id] = z3_environment.update_constraints()
+                    if condition_z3_constraint is not None and edge.negate is not None:
```

Comment:
Overall, you should not perform unnecessary refactoring. Revert the changes in this file, and see my comments below.